### DISCUSSION TITLE
How to translate AJV validation messages?

### DISCUSSION THREAD

**Discussion by gtamas** [loopbackio/loopback-next#7415] [Q&A]
I know this can be done using i18n-ajv.

So something like:

// application.ts

```
this.bind(RestBindings.REQUEST_BODY_PARSER_OPTIONS).to({
      validation: {
        ajvErrorTransformer: (errors: any[]): any[] => {
          localize.en(errors);
          return errors;
        },
      },
    });
```

This is called on every request. However, how do I detect the current locale? The request object doesn't get passed here..

**Comment by gtamas** [comment]
@bajtos @AaronJan 
Any help please?

I found this old issue. That's where this solution was proposed.

https://github.com/strongloop/loopback-next/issues/2146

However, I don't see how to detect the locale here. How can I transform the error message if I don't know the current locale.

**Reply by AaronJan** [reply]
Hi @gtamas , I did a little research but can't come up with any actionable solution right now...

I think the best way maybe is pass the Request object (or other context objects) to [this function](https://github.com/strongloop/loopback-next/blob/669e2350cd483c81bcd9dcc496989d5012c3feab/packages/rest/src/validation/request-body.validator.ts#L40) and all the way to the `ajvErrorTransformer`.

Or, we could find a way to add some contexts to the `ajv.ErrorObject` by customize validation logic (don't know how, haven't used LB4 recently so I basically forgot everything), then we can identify them in the `ajvErrorTransformer` function.

**Reply by gtamas** [reply]
Thanks for looking into this.

@bajtos 
@raymondfeng 
Any ideas? AJV is kinda everywhere, all models have validations. I guess i18n support would be important.
